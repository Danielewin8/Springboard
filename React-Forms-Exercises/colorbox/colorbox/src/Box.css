h1 {
    text-align: center;
}

.box-container {
    display: flex;
    flex-direction: row;
}

.box {
    margin: 10px;
}

.delete-btn {
    display: flex;
    flex-direction: row;
}