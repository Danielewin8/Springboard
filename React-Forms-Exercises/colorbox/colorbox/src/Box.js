import React from "react";
import './Box.css';

const Box = ({ backgroundColor, width, height, deleteBox }) => {
    width = `${width}px`
    height = `${height}px`


    return (
        <div>
            <div className="box" style={{
                backgroundColor: backgroundColor,
                width: width,
                height: height,
            }}>
            </div>
            <button className="delete-btn" onClick={deleteBox}>X</button>
        </div>
    )
}

export default Box;