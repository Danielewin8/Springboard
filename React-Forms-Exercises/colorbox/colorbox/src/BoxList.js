import React, { useState } from "react";
import Box from "./Box";
import NewBoxForm from "./NewBoxForm";

const BoxList = () => {
    const INITIAL_STATE = [
        { backgroundColor: "red", width: 100, height: 100},
        { backgroundColor: "blue", width: 100, height: 100},
        { backgroundColor: "green", width: 100, height: 100}
    ]
    const [boxes, setBoxes] = useState(INITIAL_STATE)

    const addBox = (newBox) => {
        setBoxes(boxes => [...boxes, { ...newBox }])
    }
    const deleteBox = (e) => e.target.parentElement.remove();

    return (
        <div>
            <h1>Boxes!</h1>
            <NewBoxForm addBox={addBox} />
            <div className="box-container">
                {boxes.map(({ id, backgroundColor, width, height }) => 
                <Box
                    key={id}
                    backgroundColor={backgroundColor}
                    width={width}
                    height={height} 
                    deleteBox={deleteBox}
                />
                )}
            </div>
        </div>
    )
}

export default BoxList;