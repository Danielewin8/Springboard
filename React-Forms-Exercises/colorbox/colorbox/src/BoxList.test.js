import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { toHaveStyle } from "@testing-library/jest-dom";
import BoxList from "./BoxList";
import Box from "./Box";

it("renders without crashing", function () {
    render(<BoxList />);
});

it("matches snapshot", function () {
    const { asFragment } = render(<BoxList />);
    expect(asFragment()).toMatchSnapshot();
});

test('should check length of box array after adding box', () => {
  // Render the BoxList component
  const { getByLabelText, getByText, container } = render(<BoxList />);

  // Simulate entering values in the form
  const colorInput = getByLabelText('Color');
  const widthInput = getByLabelText('Width');
  const heightInput = getByLabelText('Height');
  const addButton = getByText('Add Box');

  fireEvent.change(colorInput, { target: { value: 'purple' } });
  fireEvent.change(widthInput, { target: { value: '100' } });
  fireEvent.change(heightInput, { target: { value: '100' } });

  // Get the initial boxes count
  const initialBoxesCount = container.querySelectorAll('.box-container > div').length;

  // Simulate submitting the form
  fireEvent.click(addButton);

  // Get the updated boxes count
  const updatedBoxesCount = container.querySelectorAll('.box-container > div').length;

  // Assert that the length has increased by 1
  expect(updatedBoxesCount).toBe(initialBoxesCount + 1);
});

test('should check length of box array after adding and removing a box', () => {
    // Render the BoxList component
    const { getByLabelText, getByText, container, getAllByRole } = render(<BoxList />);
  
    // Simulate entering values in the form
    const colorInput = getByLabelText('Color');
    const widthInput = getByLabelText('Width');
    const heightInput = getByLabelText('Height');
    const addButton = getByText('Add Box');
  
    fireEvent.change(colorInput, { target: { value: 'purple' } });
    fireEvent.change(widthInput, { target: { value: '100' } });
    fireEvent.change(heightInput, { target: { value: '100' } });
  
    // Get the initial boxes count
    const initialBoxesCount = container.querySelectorAll('.box-container > div').length;
  
    // Simulate submitting the form
    fireEvent.click(addButton);
  
    // Get the updated boxes count
    const updatedBoxesCount = container.querySelectorAll('.box-container > div').length;
  
    // Assert that the length has increased by 1
    expect(updatedBoxesCount).toBe(initialBoxesCount + 1);
  
    // Get all 'X' buttons and find the one corresponding to the newly added box
    const deleteButtons = getAllByRole('button', { name: 'X' });
    const deleteButton = deleteButtons[deleteButtons.length - 1];
  
    // Simulate removing the newly added box
    fireEvent.click(deleteButton);
  
    // Get the final boxes count after removal
    const finalBoxesCount = container.querySelectorAll('.box-container > div').length;
  
    // Assert that the length is back to the initial count
    expect(finalBoxesCount).toBe(initialBoxesCount);
  });
