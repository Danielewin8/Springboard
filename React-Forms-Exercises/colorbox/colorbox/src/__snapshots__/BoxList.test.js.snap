// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Boxes!
    </h1>
    <form>
      <label
        for="color"
      >
        Color
      </label>
      <input
        id="color"
        name="backgroundColor"
        placeholder="Enter Color"
        type="text"
        value=""
      />
      <label
        for="width"
      >
        Width
      </label>
      <input
        id="width"
        name="width"
        placeholder="Enter Width"
        type="text"
        value=""
      />
      <label
        for="height"
      >
        Height
      </label>
      <input
        id="height"
        name="height"
        placeholder="Enter Height"
        type="text"
        value=""
      />
      <button>
        Add Box
      </button>
    </form>
    <div
      class="box-container"
    >
      <div>
        <div
          class="box"
          style="background-color: red; width: 100px; height: 100px;"
        />
        <button
          class="delete-btn"
        >
          X
        </button>
      </div>
      <div>
        <div
          class="box"
          style="background-color: blue; width: 100px; height: 100px;"
        />
        <button
          class="delete-btn"
        >
          X
        </button>
      </div>
      <div>
        <div
          class="box"
          style="background-color: green; width: 100px; height: 100px;"
        />
        <button
          class="delete-btn"
        >
          X
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
